### PR TITLE
Improved server detection / gsid introduced

### PR DESCRIPTION
**Pull request by annando**
This PR introduces a new field in the "gserver" table that indicate how the server data was gathered. Also new fields for the global server id ("gsid") are introduced to the contact tables.

### CODE REVIEW

File: database.sql [hunk 1]
```diff
@@ -1,9 +1,37 @@
...
 -- Friendica 2020.06-dev (Red Hot Poker)
--- DB_UPDATE_VERSION 1348
+-- DB_UPDATE_VERSION 1350
 -- ------------------------------------------
 
 
+--
+-- TABLE gserver
+--
+CREATE TABLE IF NOT EXISTS `gserver` (
```
Comment (MrPetovan) [inline database.sql:10]: Can you please move back this definition in the table name alphabetical order it was in before?

File: static/dbstructure.config.php [hunk 2]
```diff
@@ -54,11 +54,40 @@
 use Friendica\Database\DBA;
 
 if (!defined('DB_UPDATE_VERSION')) {
-	define('DB_UPDATE_VERSION', 1349);
+	define('DB_UPDATE_VERSION', 1350);
 }
 
 return [
 	// Side tables
+	"gserver" => [
```
Comment (MrPetovan) [inline static/dbstructure.config.php:62]: Can you please move back this definition in the table name alphabetical order it was in before?

File: database.sql (hunk 1 above)
Comment (annando) [reply database.sql:10]: Sadly this is not possible. Due to the foreign keys, the tables where foreign keys rely upon needs to be created first.

File: static/dbstructure.config.php (hunk 2 above)
Comment (annando) [reply static/dbstructure.config.php:62]: See above.

Possibly we can change this foreign key creation in a way that we always perform this after the table had been created. But vurrently this is the only way. I don't like it as well.

File: database.sql (hunk 1 above)
Comment (MrPetovan) [reply database.sql:10]: Got it.

File: static/dbstructure.config.php (hunk 2 above)
Comment (MrPetovan) [reply static/dbstructure.config.php:62]: Related to [this comment](https://github.com/friendica/friendica/pull/8678#discussion_r429268068), I'd rather have a programmatic way of ordering table creation than relying on the implicit order of this array.

Comment (annando) [reply static/dbstructure.config.php:62]: Especially we could have some circular dependancies, I guess. Then it could get really funny :-)

Comment (MrPetovan) [reply static/dbstructure.config.php:62]: `mysqldump` has an option to create all the indices after tables have been created to avoid such an issue.

**Pull request merged**